flume-ng-sinks/flume-dataset-sink/src/main/java/org/apache/flume/sink/kite/DatasetSink.java

```diff
@@ -219,52 +249,47 @@ public synchronized void stop() {
   @Override
   public Status process() throws EventDeliveryException {
     if (writer == null) {
-      try {
-        this.writer = newWriter(login, target);
-      } catch (DatasetException e) {
-        // DatasetException includes DatasetNotFoundException
-        throw new EventDeliveryException(
-          "Cannot write to " + getName(), e);
-      }
+      this.writer = newWriter();
     }
 
-    // handle file rolling
-    if ((System.currentTimeMillis() - lastRolledMs) / 1000 > rollIntervalS) {
-      // close the current writer and get a new one
-      writer.close();
-      this.writer = newWriter(login, target);
-      this.lastRolledMs = System.currentTimeMillis();
-      LOG.info("Rolled writer for " + getName());
+    if (shouldRoll()) {
+      rollWriter(true, true);
     }
 
     Channel channel = getChannel();
-    Transaction transaction = null;
     try {
       long processedEvents = 0;
 
-      transaction = channel.getTransaction();
-      transaction.begin();
+      // TODO: Can the same Sink object be called from multiple threads
```

Comment:
It appears that way, but that sounds crazy (stopping in another thread when one might be running it?) and should be done by signalling the SinkRunner to stop (after process finishes), then stopping/joining the thread, and finally by stopping the processor (which could be done by SinkRunner when it is notified). Let's ask for clarification from Hari to make sure.